enviPath_python/objects.py

```diff
@@ -2851,664 +3056,4265 @@ def parse(cls, data_string):
+class SourceScenarioAdditionalInformation(AdditionalInformation):
+    """
+    Creates a source scenario additional information object.
+
+    This class represents additional information about the source scenario represented as the scource scenario ID.
+    """
+    name = "sourcescenario"
+    mandatories = ["sourcescenario"]
+
+    # Setter
+    def set_sourcescenario(self, value):
+        """
+        Sets the source scenario.
+
+        :param value: The source scenario as scenario ID.
+        :type value: str
+        """
+        self.params["sourcescenario"] = value
+
+    # Getter
+    def get_sourcescenario(self):
+        """
+        Gets the source scenario.
+
+        :return: The source scenario, or None if not set.
+        :rtype: str or None
+        """
+        return self.params.get("sourcescenario", None)
+
+    # Parser
+    @classmethod
+    def parse(cls, data_string):
+        """
+        Parses the data_string to create a SourceScenarioAdditionalInformation instance.
+
+        :param data_string: String containing the source scenario.
+        :type data_string: str
+        :return: SourceScenarioAdditionalInformation instance.
+        :rtype: SourceScenarioAdditionalInformation
+        """
+        res = {"sourcescenario": data_string}
+        return cls(**res)
+    
+class SpikeCompoundAdditionalInformation(AdditionalInformation):
+    """
+    Creates a Spike Compound additional information object.
+
+    This class represents additional information about spike compounds. It takes either an existing compound or creates a new one using a smile.
+    """
+    name = "spikecompound"
+    mandatories = []
 
     # Setter
-    def set_referringscenario(self, value):
-        self.params["referringscenario"] = value
+    def set_spikeComp(self, value):
+        """
+        Sets the spike compound information. 
+
+        :param value: The spike compound information in the form of an existing compound ID or a smile (to create a new compound).
+        :type value: str
+        """
+        self.params["spikeComp"] = value
 
     # Getter
-    def get_referringscenario(self):
-        return self.params.get("referringscenario", None)
+    def get_spikeComp(self):
+        """
+        Gets the spike compound information.
+
+        :return: The spike compound information, or None if not set.
+        :rtype: str or None
+        """
+        return self.params.get("spikeComp", None)
 
     # Parser
     @classmethod
     def parse(cls, data_string):
-        return cls._parse_default(data_string, ['referringscenario'])
+        """
+        Parses the data_string to create a SpikeCompoundAdditionalInformation instance.
+
+        :param data_string: A string containing spike compound data.
+        :type data_string: str
+        :return: SpikeCompoundAdditionalInformation instance.
+        :rtype: SpikeCompoundAdditionalInformation
+        """
+        res = {"spikeComp": data_string}
+        return cls(**res)
```

Comment:
We cannot get the info of the spike compound if the only data stored on "value" is the compound name